CommonData/column.cpp

```diff
@@ -1648,24 +1662,43 @@ void Column::labelsOrderByValue(bool doDbUpdateEtc)
 	if(replaceAllDoubles)
 		replaceDoublesTillLabelsRowWithLabels(labelsTempCount());
 	
-	doublevec				asc			= valuesNumericOrdered();
-	size_t					curMax		= asc.size()+1;
-	std::map<double, int>	orderMap;
-	
-	for(size_t i=0; i<asc.size(); i++)
-		orderMap[asc[i]] = i;
-	
-	//and now to write them back into the data
-	for(Label * label : _labels)
+	doublevec asc = valuesNumericOrdered();
+
+	if (asc.empty())
```

Comment:
So with this change it either orders the numeric ones *or* the non-numerics if there are no numerics?

Why not order both?